bindings/ergo-lib-ios/Tests/ErgoLibTests/RestNodeApiTests.swift

```diff
@@ -1,49 +1,103 @@
             })
         waitForExpectations(timeout: 5, handler: nil)
     }
-
-    func testGetInfoAbort() throws {
-        let nodeConf = try NodeConf(withAddrString: "127.0.0.1:9053")
+    
+    func testGetNipopowProofByHeaderAbort() throws {
+        let nodeConf = try NodeConf(withAddrString: "213.239.193.208:9053")
         let restNodeApi = try RestNodeApi()
-        let handle = try restNodeApi.getInfo(nodeConf: nodeConf,
-            closure: { (res: Result<NodeInfo, Error>) -> () in 
+        let blockHeaders = try HeaderTests.generateBlockHeadersFromJSON()
+        let handle = try restNodeApi.getNipopowProofByHeaderId(
+            nodeConf: nodeConf,
+            minChainLength: UInt32(3),
+            suffixLen: UInt32(2),
+            headerId: blockHeaders.get(index: UInt(0))!.getBlockId(),
+            closure: { (res: Result<NipopowProof, Error>) -> () in
                 XCTFail("this should not be called")
             })
         handle.abort()
     }
-
-    // need macOS 12.0 on Github GA
-    func testGetInfoAsync() throws {
-        let nodeConf = try NodeConf(withAddrString: "127.0.0.1:9053")
+    
+    func testGetNipopowProofByHeaderIdAsync() async throws {
+        let nodeConf = try NodeConf(withAddrString: "213.239.193.208:9053")
+        let restNodeApi = try RestNodeApi()
+        let blockHeaders = try HeaderTests.generateBlockHeadersFromJSON()
+        let proof = try await restNodeApi.getNipopowProofByHeaderIdAsync(
+            nodeConf: nodeConf,
+            minChainLength: UInt32(3),
+            suffixLen: UInt32(2),
+            headerId: blockHeaders.get(index: UInt(0))!.getBlockId()
+        )
+        XCTAssertNoThrow(try proof.toJSON()!)
+        
+        // test of re-using of tokio runtime
+        let proofNew = try await restNodeApi.getNipopowProofByHeaderIdAsync(
+            nodeConf: nodeConf,
+            minChainLength: UInt32(3),
+            suffixLen: UInt32(2),
+            headerId: blockHeaders.get(index: UInt(0))!.getBlockId()
+        )
+        XCTAssertNoThrow(try proofNew.toJSON()!)
+    }
+    
+    func testPeerDiscoveryNonAsync() throws {
+        let expectation = self.expectation(description: "peerDiscovery")
+        let restNodeApi = try RestNodeApi()
+        let _ = try restNodeApi.peerDiscovery(
+            seeds: getSeeds(),
+            maxParallelReqs: UInt16(30),
+            timeoutSec: UInt32(2),
+            closure: { (res: Result<CStringCollection, Error>) -> () in
+                switch res {
+                    case .success(let peers):
+                        XCTAssert(peers.getLength() > 0)
+                        break
+                    case .failure(let error):
+                        XCTFail(error.localizedDescription)
+                }
+                expectation.fulfill()
+            })
+        waitForExpectations(timeout: 60, handler: nil)
+    }
+    
+    func testPeerDiscoveryAsync() async throws {
```

Comment:
The async tests mistakenly result in 'pass' when using `Task` in the unit tests. The fix is to make the entire function async and avoid making calls to `Task`.